### ISSUE TITLE
False positive no-member when using cls to instantiate an object in super() 

### ISSUE THREAD

**Issue by yanqd0** [PyCQA/pylint#4241]
### Steps to reproduce

Given a file `a.py`:
```python
# pylint: disable=too-few-public-methods, missing-module-docstring
# pylint: disable=missing-class-docstring, missing-function-docstring
# pylint: disable=invalid-name
class A:
    a = 1

    @classmethod
    def create(cls):
        return cls()


class B(A):
    b = 2

    @classmethod
    def create(cls):
        return super().create()


b = B.create()
print(b.b)
```

### Current behavior

Result of `pylint a.py`:

```
$ pylint a.py
************* Module a
a.py:21:6: E1101: Instance of 'A' has no 'b' member; maybe 'a'? (no-member)
```

### Expected behavior

No error or warning.

### pylint --version output

Result of `pylint --version` output:
```
pylint 2.7.2
astroid 2.5.1
Python 3.8.5 (default, Jan 27 2021, 15:41:15) 
[GCC 9.3.0]
```

**Comment by hippo91** [comment]
@yanqd0 thanks for the report.

**Comment by jacobtylerwalls** [comment]
There are a lot of tickets for `no-member` and subclassing: #4241, #3805, #3143, #3162, and more. I haven't carefully checked to see whether they're all more or less a result of the same root cause.

**Comment by jacobtylerwalls** [comment]
They probably are: see https://github.com/PyCQA/astroid/issues/457

**Comment by jacobtylerwalls** [comment]
Duplicate of https://github.com/PyCQA/pylint/issues/981 and https://github.com/PyCQA/astroid/issues/457
